Topics/Tech_Stacks/PostgreSQL_psycopg2.md

```diff
@@ -45,85 +46,252 @@ A quick and basic runover of PostgreSQL.
 ```
-This gives you a smaller table that only contains the columns "col_name1" and "col_name2" from your selected table in your relation. 
-
-You can also merge two different tables by changing the FROM clause to contain 2 tables:
+In this example, the output is a smaller table that only contains the columns `first_name` and `last_name` of each row from the table `players`:
+| first_name | last_name |
+| ---------- | --------- |
+| Stephen | Curry |
+| Erling | Haaland |
+| Kylian | Mbappé |
+
+You can also get the Cartesian product of two or more tables by listing the tables in the `FROM` clause:
 ```
-    SELECT col1_table1, col2_table2 FROM table1_name, table2_name;
+SELECT * FROM sports, teams;
 ```
-This operation, first, performs a cross product between table1 and table2, where every row of one table is matched with every other row of the other table. Then it selects the data present in the new cross-products table from columns col1_table1 and col2_table2.
+In this query, each row of table `sports` is combined with each row of the table `teams`, resulting in a combination of all rows:
+| sport_id | sport_name | team_id | team_name | sport_id |
+| -------- | ---------- | ------- | --------- | -------- |
+| 1 | basketball | 1 | Toronto Raptors | 1 |
+| 1 | basketball | 2 | San Antonio Spurs | 1 |
+| 1 | basketball | 3 | Golden State Warriors | 1 |
+| 1 | basketball | 4 | Manchester City F.C. | 2 |
+| 1 | basketball | 5 | Paris Saint-Germain F.C. | 2 |
+| 2 | soccer | 1 | Toronto Raptors | 1 |
+| 2 | soccer | 2 | San Antonio Spurs | 1 |
+| 2 | soccer | 3 | Golden State Warriors | 1 |
+| 2 | soccer | 4 | Manchester City F.C. | 2 |
+| 2 | soccer | 5 | Paris Saint-Germain F.C. | 2 |
+| 3 | baseball | 1 | Toronto Raptors | 1 |
+| 3 | baseball | 2 | San Antonio Spurs | 1 |
+| 3 | baseball | 3 | Golden State Warriors | 1 |
+| 3 | baseball | 4 | Manchester City F.C. | 2 |
+| 3 | baseball | 5 | Paris Saint-Germain F.C. | 2 |
+
+However, not all rows of the output are meaningful. For example, the fourth row associates basketball with Manchester City F.C., which is a soccer team. To filter out the nonsensical combinations, we can use the `WHERE` clause:
+```
+SELECT * FROM sports, teams WHERE sports.sport_id = teams.sport_id;
+```
+This clause allows you to retrieve only the rows that meet the specified conditions. The output of this query would be:
+| sport_id | sport_name | team_id | team_name | sport_id |
+| -------- | ---------- | ------- | --------- | -------- |
+| 1 | basketball | 1 | Toronto Raptors | 1 |
+| 1 | basketball | 2 | San Antonio Spurs | 1 |
+| 1 | basketball | 3 | Golden State Warriors | 1 |
+| 2 | soccer | 4 | Manchester City F.C. | 2 |
+| 2 | soccer | 5 | Paris Saint-Germain F.C. | 2 |
+
+Although the query is now more targeted and relevant, the two `sport_id` columns are redundant. To get rid of the redundancy, one solution is to change the columns in the `SELECT` clause so that only one of the `sport_id` columns is retrieved:
```

Comment:
What is meant by relevant? Maybe you can specify a goal and talk about how it is fulfilled by this join operation. For example, the goal could be to see all the players of each sport.